src/tracks/SampleTrack.cpp

```diff
@@ -109,6 +109,19 @@ SampleTCO::SampleTCO( Track * _track ) :
 
 
 
+SampleTCO::SampleTCO(const SampleTCO& orig) :
+	SampleTCO(orig.getTrack())
+{
+	// TODO: This creates a new SampleBuffer for the new TCO, eating up memory
+	// & eventually causing performance issues. Letting tracks share buffers
+	// when they're identical would fix this, but isn't possible right now.
+	m_sampleBuffer = new SampleBuffer(*orig.m_sampleBuffer);
+	m_isPlaying = orig.m_isPlaying;
+}
```

Comment:
The constructor delegated to here initialises `m_sampleBuffer` with a newly allocated, default-constructed `SampleBuffer`. It is then leaked when this constructor overwrites it.

Reply:
Hopefully fixed in latest commit